contracts/sky/src/query.rs

```diff
@@ -0,0 +1,207 @@
+use cosmwasm_std::{
+    Storage, Api, Querier, Extern, StdResult, StdError, debug_print,
+};
+use cosmwasm_math_compat::Uint128;
+use secret_toolkit::utils::Query;
+use shade_protocol::{
+    contract_interfaces::{
+        sky::sky::{QueryAnswer, Config, ViewingKeys, SelfAddr},
+        mint::mint::{QueryMsg, self},
+        dex::{dex::pool_take_amount, sienna::{PairInfoResponse, PairQuery, TokenType, PairInfo},},
+    snip20,
+    },
+    utils::storage::plus::ItemStorage,
+};
+
+pub fn config<S: Storage, A: Api, Q: Querier>(
+    deps: &Extern<S, A, Q>
+) -> StdResult<QueryAnswer> {
+    Ok(QueryAnswer::Config {
+        config: Config::load(&deps.storage)?,
+    })
+}
+
+pub fn market_rate<S: Storage, A: Api, Q: Querier>(
+    deps: &Extern<S, A, Q>
+) -> StdResult<QueryAnswer> {
+    let config: Config = Config::load(&deps.storage)?;
+
+    //Query mint contract
+    let mint_info: mint::QueryAnswer = QueryMsg::Mint{
+        offer_asset: config.shd_token.contract.address.clone(),
+        amount: Uint128::new(100000000), //1 SHD
+    }.query(
+        &deps.querier,
+        config.mint_addr.code_hash.clone(),
+        config.mint_addr.address.clone(),
+    )?;
+    let mut mint_price: Uint128 = Uint128::new(0); // SILK/SHD
+    match mint_info{
+        mint::QueryAnswer::Mint {
+            asset: _,
+            amount,
+        } => {
+            mint_price = amount.checked_mul(Uint128::new(100)).unwrap(); // times 100 to make it have 8 decimals
```

Comment:
Since the function returns StdResult, you can get away with doing like so:
```suggestion
            mint_price = amount.checked_mul(Uint128::new(100))?; // times 100 to make it have 8 decimals
```